controls/SV-230251.rb

```diff
@@ -1,83 +1,75 @@
+
+If the MACs entries in the "opensshserver.config" file have any hashes other than shown here, the order differs from the example above, or they are missing or commented out, this is a finding.'
+
+  desc "fix", 'Configure the RHEL 8 SSH server to use only MACs employing FIPS 140-2-approved algorithms by updating the "/etc/crypto-policies/back-ends/opensshserver.config" file with the following line:
 
-    If the MACs entries in the \"opensshserver.config\" file have any hashes
-other than \"hmac-sha2-512\" and \"hmac-sha2-256\", the order differs from the
-example above, they are missing, or commented out, this is a finding.
-  "
-  desc  'fix', "
-    Configure the RHEL 8 SSH server to use only MACs employing FIPS
-140-2-approved algorithms by updating the
-\"/etc/crypto-policies/back-ends/opensshserver.config\" file with the following
-line:
+    -oMACS=hmac-sha2-512,hmac-sha2-256,hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com
 
-    -oMACS=hmac-sha2-512,hmac-sha2-256
+    A reboot is required for the changes to take effect.'
 
-    A reboot is required for the changes to take effect.
-  "
   impact 0.5
-  tag severity: 'medium'
-  tag gtitle: 'SRG-OS-000250-GPOS-00093'
-  tag satisfies: ['SRG-OS-000250-GPOS-00093', 'SRG-OS-000393-GPOS-00173',
-'SRG-OS-000394-GPOS-00174', 'SRG-OS-000125-GPOS-00065']
-  tag gid: 'V-230251'
-  tag rid: 'SV-230251r743937_rule'
-  tag stig_id: 'RHEL-08-010290'
-  tag fix_id: 'F-32895r743936_fix'
-  tag cci: ['CCI-001453']
-  tag nist: ['AC-17 (2)']
-
-  if virtualization.system.eql?('docker') && !file('/etc/sysconfig/sshd').exist?
-    impact 0.0
-    describe "Control not applicable - SSH is not installed within containerized RHEL" do
-      skip "Control not applicable - SSH is not installed within containerized RHEL"
-    end
-  else
-    describe parse_config_file('/etc/crypto-policies/back-ends/opensshserver.config') do
-      its('CRYPTO_POLICY') { should_not be_nil }
-    end
 
-    crypto_policy = parse_config_file('/etc/crypto-policies/back-ends/opensshserver.config')['CRYPTO_POLICY']
+  tag severity: "medium"
+  tag gtitle: "SRG-OS-000250-GPOS-00093"
+  tag satisfies: ["SRG-OS-000250-GPOS-00093", "SRG-OS-000393-GPOS-00173", "SRG-OS-000394-GPOS-00174", "SRG-OS-000125-GPOS-00065"]
+  tag gid: "V-230251"
+  tag rid: "SV-230251r917870_rule"
+  tag stig_id: "RHEL-08-010290"
+  tag fix_id: "F-32895r917869_fix"
+  tag cci: ["CCI-001453"]
+  tag nist: ["AC-17 (2)"]
+
+  # Check if SSH is installed within containerized RHEL
+  only_if("SSH is not installed within containerized RHEL. Therefore, this requirement is not applicable.", impact: 0.0) do
+    !(virtualization.system.eql?("docker") && !file("/etc/sysconfig/sshd").exist?)
+  end
+
+  # Define the required algorithms
+  required_algorithms = input('openssh_server_required_algorithms')
+
+  # TODO: make a simple resource for this based off 'login_defs' or 'yum' as a model
+
```

Comment:
@em-c-rod   # make a simple resource for this based off 'login_defs' or 'yum' as a model